pkg/daemon/controller/controller.go

```diff
@@ -232,6 +244,12 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	// if the node has never been configured, cordon it until the configuration completes
```

Comment:
Wouldn't you always cordon the node even it has been configured before?

We have also discussed if we need to be draining the node in addition. Do we need to do that?

Reply:
There isn't currently a mechanism in the code to identify a faulty service.

The closest thing to that was removed from this PR due to a previous comment, draining and cordoning if WICD fails to successfully reconcile a service multiple times in a row.

Reply:
To be clear, by faulty service, I mean a service that was tampered with and was corrected by WICD.